core/src/main/java/org/opensearch/sql/planner/physical/UnnestOperator.java

```diff
@@ -0,0 +1,252 @@
+    return visitor.visitUnnest(this, context);
+  }
+
+  @Override
+  public List<PhysicalPlan> getChild() {
+    return Collections.singletonList(input);
+  }
+
+  @Override
+  public boolean hasNext() {
+    return input.hasNext() || flattenedResult.hasNext();
+  }
+
+
+  @Override
+  public ExprValue next() {
+    if (!flattenedResult.hasNext()) {
+      result.clear();
+      nonNestedFields.clear();
+
+      ExprValue inputValue = input.next();
+      generateNonNestedFieldsMap(inputValue);
+      for (String field : fields) {
+        result = flatten(field, inputValue, result, true);
+      }
+
+      if (result.isEmpty()) {
+        return new ExprTupleValue(new LinkedHashMap<>());
+      }
+
+      for (String nonNestedField : nonNestedFields) {
+        result = flatten(nonNestedField, inputValue, result, false);
+      }
+
+      flattenedResult = result.listIterator();
+    }
+    return new ExprTupleValue(new LinkedHashMap<>(flattenedResult.next()));
+  }
+
+  /**
+   * Generate list of non-nested fields that are in inputMap, but not in the member variable
+   * fields list.
+   * @param inputMap : Row to parse non-nested fields.
+   */
+  public void generateNonNestedFieldsMap(ExprValue inputMap) {
+
+    for (Map.Entry<String, ExprValue> inputField : inputMap.tupleValue().entrySet()) {
+      boolean foundNestedField = false;
+      for (String field : this.fields) {
+        if ((field).split("\\.")[0].equalsIgnoreCase(inputField.getKey())) {
+          foundNestedField = true;
+          break;
+        }
+      }
+
+      if (!foundNestedField) {
+        boolean nestingComplete = false;
+        String nonNestedField = inputField.getKey();
+        ExprValue currentObj = inputField.getValue();
+        while (!nestingComplete) {
+          if (currentObj instanceof ExprTupleValue) {
+            var it = currentObj.tupleValue().entrySet().iterator().next();
+            currentObj = it.getValue();
+            nonNestedField += "." + it.getKey();
+          } else if (currentObj instanceof ExprCollectionValue) {
+            currentObj = currentObj.collectionValue().get(0);
+          } else {
+            nestingComplete = true;
+          }
+        }
+        this.nonNestedFields.add(nonNestedField);
+      }
+    }
+  }
+
+
+  /**
+   * Simplifies the structure of row's source Map by flattening it,
+   * making the full path of an object the key
+   * and the Object it refers to the value.
+   *
+   * <p>Sample input:
+   * keys = ['comments.likes']
+   * row = comments: {
+   * likes: 2
+   * }
+   *
+   * <p>Return:
+   * flattenedRow = {comment.likes: 2}
+   *
+   * @param nestedField : Field to query in row.
+   * @param row : Row returned from OS.
+   * @param prevList : List of previous nested calls.
+   * @param supportArrays : When false we do not need to execute a cross join.
+   * @return : List of nested select items or cartesian product of nested calls.
+   */
+  @SuppressWarnings("unchecked")
```

Comment:
can we avoid suppressing?

Reply:
Fixed in [a574cb6d78483e1b329d5db5a9fe57d03ffa4191](https://github.com/Bit-Quill/opensearch-project-sql/pull/243/commits/a574cb6d78483e1b329d5db5a9fe57d03ffa4191)